packages/cli/src/models/network/NetworkController.ts

```diff
@@ -226,12 +227,16 @@ export default class NetworkController {
 
   // Contract model || SolidityLib model
   public async uploadSolidityLibs(libs: Contract[]): Promise<void> {
-    await allPromisesOrError(libs.map(lib => this._uploadSolidityLib(lib)));
+    // Libs may have dependencies, so deploy them in order
+    for (let i = 0; i < libs.length; i++) {
+      await this._uploadSolidityLib(libs[i]);
```

Comment:
Is it necessary to wait for each deployment to be mined before the next one happens? Execution order should be guaranteed by the sender account's nonce field, and this approach will make deployment much slower.

That said, such a deployment is a rare scenario, and this _is_ safer in that a failing transaction with halt the whole deployment process, as opposed to potentially causing in-flight transactions to fail.

Reply:
Nevermind, we need for the current library deployment to actually go through because future libraries may depend on it, and need its address to link to it.

Reply:
Yes. It is a dependency tree.